### PR TITLE
Drop jinja2 in trained model serialization tests

### PR DESCRIPTION
**Pull request by micmn**
It may need some cleanup but this should do it.

@karlnapf #3537

### CODE REVIEW

**Comment by vigsterkr** [comment]
@micmn from appveyor:
```[00:49:40]     15>c:\projects\shogun\tests\unit\base\trained_model_serialization_unittest.cc(127): fatal error C1083: Cannot open include file: 'trained_model_serialization_unittest.h': No such file or directory [C:\projects\shogun\build\tests\unit\shogun-unit-test.vcxproj]```

File: tests/unit/base/trained_model_serialization_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+			SG_SERROR("Unsupported problem type: %d\n", pt);
+			FAIL();
+		}
+
+		SG_REF(train_feats)
+		SG_REF(test_feats)
+		SG_REF(train_labels)
+	}
+
+	bool serialize_machine(
```
Comment (karlnapf) [inline tests/unit/base/trained_model_serialization_unittest.cc]: shall we put this into a helper, this kind of stuff is used in multiple places in the tests

File: tests/unit/base/trained_model_serialization_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		bool load_success = cmachine->load_serializable(file);
+
+		file->close();
+		SG_FREE(file);
+		int delete_success = unlink(filename.c_str());
+
+		return load_success && (delete_success == 0);
+	}
+
+	CDenseFeatures<float64_t> *train_feats, *test_feats;
```
Comment (karlnapf) [inline tests/unit/base/trained_model_serialization_unittest.cc]: could we template the features as well?

**Comment by shubham808** [comment]
@micmn I need to extend this for pausing tests so i am happy to complete this for u :)

**Comment by karlnapf** [comment]
@micmn thx for the patch. We will cherry pick your commit and move on from there if you dont mind?

**Comment by micmn** [comment]
Yes, of course :)

**Comment by karlnapf** [comment]
@micmn we have trouble fixing this error. Would you mind having a look? :)

**Comment by micmn** [comment]
green light!

hello Microsoft :o

**Comment by karlnapf** [comment]
Nice! Thanks!

**Pull request merged**